private/pkg/app/appproto/appprotoexec/binary_handler.go

```diff
@@ -67,13 +70,19 @@ func (h *binaryHandler) Handle(
 	}
 	responseBuffer := bytes.NewBuffer(nil)
 	stderrWriteCloser := newStderrWriteCloser(container.Stderr(), h.pluginPath)
-	if err := h.runner.Run(
-		ctx,
-		h.pluginPath,
+	runOptions := []command.RunOption{
 		command.RunWithEnv(app.EnvironMap(container)),
 		command.RunWithStdin(bytes.NewReader(requestData)),
 		command.RunWithStdout(responseBuffer),
 		command.RunWithStderr(stderrWriteCloser),
+	}
+	if h.pluginArgs != nil {
+		runOptions = append(runOptions, command.RunWithArgs(h.pluginArgs...))
+	}
```

Comment:
It is best practice to examine the length of the slice, instead of testing if nil. It is rare that you want or need to distinguish between empty and nil slice (and can also be brittle).
```suggestion
	if len(h.pluginArgs) > 0 {
		runOptions = append(runOptions, command.RunWithArgs(h.pluginArgs...))
	}
```